lib/std/maybes.nim

```diff
@@ -0,0 +1,34 @@
+type Maybe*[T] = object
+  valueImpl*: T
+
+proc maybe*[T](a: T): auto =
+  ## Allows chains of field-access and indexing where the LHS can be nil.
+  ## This simplifies code by reducing need for if-else branches around intermediate
+  ## maybe nil values.
+  runnableExamples:
+    type Foo = ref object
+      x1: string
+      x2: Foo
+    var f: Foo
+    assert f.maybe.x2.x1[] == ""
+  Maybe[T](valueImpl: a)
+
+template `.`*(a: Maybe, b): untyped =
```

Comment:
`{.experimental: "dotOperators".}` is not needed anymore?

Reply:
that's for macros, not templates according to current implementation ; we can fix either spec or implementation but might as well fix the spec to match the implementation since existing code already relies on this behavior (templates being less powerful than macros, IMO it's ok to keep allowing template dotOperators without `--experimental:dotOperators`

Reply:
It should require ``{.experimental: "dotOperators".}``.

Reply:
done; and filed https://github.com/nim-lang/Nim/issues/13034